zcash_test_vectors/transaction.py

```diff
@@ -462,6 +477,11 @@ def __init__(self, rand, consensus_branch_id):
             # v^balanceSapling is defined to be 0.
             self.valueBalanceSapling = 0
 
+    def init_orchard(self, rand, flip_coins):
+        have_transparent_in = (flip_coins >> 0) % 2
+        have_orchard = (flip_coins >> 3) % 2
+        is_coinbase = (not have_transparent_in) and (flip_coins >> 4) % 2
```

Comment:
Same bug, `have_transparent_in` and `is_coinbase` must be consistent for the entire transaction.